cli/azd/cmd/auth_serve.go

```diff
@@ -0,0 +1,141 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/pkg/auth"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/spf13/cobra"
+)
+
+// TokenResponse defines the structure of the token response.
+type TokenResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpiresOn   int64  `json:"expires_on"`
+}
+
+// tokenHandler handles token requests.
+func (serve *serveAction) tokenHandler(w http.ResponseWriter, r *http.Request) {
+	resource := r.URL.Query().Get("resource")
+	if resource == "" {
+		resource = "https://management.azure.com/"
+	}
+
+	fmt.Printf("Received request for resource: %s\n", resource)
+
+	ctx := context.Background()
+	var cred azcore.TokenCredential
+
+	cred, err := serve.credentialProvider(ctx, &auth.CredentialForCurrentUserOptions{
+		NoPrompt: true,
+		TenantID: "",
+	})
+	if err != nil {
+		fmt.Printf("credentialProvider: %v", err)
+		return
+	}
+
+	token, err := cred.GetToken(ctx, policy.TokenRequestOptions{
+		Scopes: []string{resource + "/.default"},
+	})
+	if err != nil {
+		fmt.Printf("fetching token: %v", err)
+		return
+	}
+
+	res := TokenResponse{
+		AccessToken: token.Token,
+		ExpiresOn:   token.ExpiresOn.Unix(),
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(res); err != nil {
+		http.Error(w, "Failed to encode response: "+err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (serve *serveAction) start(port string) {
```

Comment:
NIT: It would be better if this was an `int` not a `string` perhaps?  If we pass in a non int I assume `ListenAndServe()` is going to return an error.